sympy/sets/sets.py

```diff
@@ -1465,12 +1465,32 @@ def reduce(args):
                 unk = [x for x in s
                        if any(other.contains(x) not in (True, False) for other in other_args)]
                 if unk:
+                    for val in other_args:
+                        if val.is_FiniteSet:
+                            # all values from val which are contained in `s`
+                            # and are non-numbers
+                            t = [x for x in s if val.contains(x) is true and not x.is_number]
```

Comment:
Yes, i guess we can remove `is true`. But i thought to maintain a 
consistency, i.e to always use `S.true` whenever `.contains` is there.

On Fri, Oct 30, 2015 at 10:21 PM, Kalevi Suominen 
<notifications@github.com> wrote:
> In sympy/sets/sets.py:
> 
> > @@ -1465,12 +1465,32 @@ def reduce(args):
> >                  unk = [x for x in s
> >                         if any(other.contains(x) not in (True, 
> False) for other in other_args)]
> >                  if unk:
> > +                    for val in other_args:
> > +                        if val.is_FiniteSet:
> > +                            # all values from val which are 
> contained in `s`
> > +                            # and are non-numbers
> > +                            t = [x for x in s if val.contains(x) 
> is true and not x.is_number]
> I guess this is ok then. Perhaps one could even say if 
> val.contains(x) omitting is true?
> 
> —
> Reply to this email directly or view it on GitHub.
>